src/audio/source_api_helper.c

```diff
@@ -0,0 +1,167 @@
+{
+	return source->num_of_bytes_processed;
+}
+
+void source_reset_num_of_processed_bytes(struct sof_source __sparse_cache *source)
+{
+	source->num_of_bytes_processed = 0;
+}
+
+enum sof_ipc_frame source_get_frm_fmt(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->frame_fmt;
+}
+
+enum sof_ipc_frame source_get_valid_fmt(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->valid_sample_fmt;
+}
+
+unsigned int source_get_rate(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->rate;
+}
+
+unsigned int source_get_channels(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->channels;
+}
+
+uint32_t source_get_buffer_fmt(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->buffer_fmt;
+}
+
+bool source_get_underrun(struct sof_source __sparse_cache *source)
+{
+	return source->audio_stream_params->underrun_permitted;
+}
+
+int source_set_valid_fmt(struct sof_source __sparse_cache *source,
+			 enum sof_ipc_frame valid_sample_fmt)
+{
+	source->audio_stream_params->valid_sample_fmt = valid_sample_fmt;
+	if (source->ops->on_audio_format_set)
+		return source->ops->on_audio_format_set(source);
+	return 0;
+}
+
+int source_set_rate(struct sof_source __sparse_cache *source, unsigned int rate)
+{
+	source->audio_stream_params->rate = rate;
+	if (source->ops->on_audio_format_set)
+		return source->ops->on_audio_format_set(source);
+	return 0;
+}
+
+int source_set_channels(struct sof_source __sparse_cache *source, unsigned int channels)
+{
+	source->audio_stream_params->channels = channels;
+	if (source->ops->on_audio_format_set)
+		return source->ops->on_audio_format_set(source);
+	return 0;
+}
+
+int source_set_buffer_fmt(struct sof_source __sparse_cache *source, uint32_t buffer_fmt)
+{
+	source->audio_stream_params->buffer_fmt = buffer_fmt;
+	if (source->ops->on_audio_format_set)
+		return source->ops->on_audio_format_set(source);
+	return 0;
+}
+
+int source_set_underrun(struct sof_source __sparse_cache *source, bool underrun_permitted)
+{
+	source->audio_stream_params->underrun_permitted = underrun_permitted;
+	if (source->ops->on_audio_format_set)
+		return source->ops->on_audio_format_set(source);
+	return 0;
+}
+
+size_t source_get_frame_bytes(struct sof_source __sparse_cache *source)
+{
+	return get_frame_bytes(source_get_frm_fmt(source),
+				source_get_channels(source));
+}
+
+size_t source_get_data_frames_available(struct sof_source __sparse_cache *source)
+{
+	return source_get_data_available(source) /
+			source_get_frame_bytes(source);
+}
+
+int source_set_params(struct sof_source __sparse_cache *source,
+		      struct sof_ipc_stream_params *params, bool force_update)
+{
+	if (source->ops->audio_set_ipc_params)
+		return source->ops->audio_set_ipc_params(source, params, force_update);
+	return 0;
+}
+
+int source_set_alignment_constants(struct sof_source __sparse_cache *source,
```

Comment:
I would also prefix with sof_ all the functions, just like you did for the struct sof_source. It'd be more future- and conflict-proof.